databricks/koalas/tests/test_dataframe.py

```diff
@@ -2397,3 +2397,50 @@ def test_mask(self):
 
         with self.assertRaisesRegex(ValueError, 'type of cond must be a DataFrame or Series'):
             kdf.mask(1)
+
+    def test_query(self):
+        kdf = ks.DataFrame(
+            {'A': range(1, 6),
+             'B': range(10, 0, -2),
+             'C': range(10, 5, -1)})
+        pdf = kdf.to_pandas()
+
+        exprs = ('A > B',
+                 'A < C',
+                 'C == B')
+        for expr in exprs:
+            self.assert_eq(kdf.query(expr), pdf.query(expr))
+
+        # test `inplace=True`
+        for expr in exprs:
+            dummy_kdf = kdf.copy()
+            dummy_pdf = pdf.copy()
+
+            pdf.query(expr, inplace=True)
+            kdf.query(expr, inplace=True)
+
+            self.assert_eq(dummy_kdf, dummy_pdf)
+
+        # invalid values for `expr`
+        invalid_exprs = (1, 1.0, (exprs[0],), [exprs[0]])
+        for expr in invalid_exprs:
+            with self.assertRaisesRegex(
+                    ValueError,
+                    'expr must be a string to be evaluated, {} given'
+                    .format(type(expr))):
+                kdf.query(expr)
+
+        # invalid values for `inplace`
+        invalid_inplaces = (1, 0, 'True', 'False')
+        for inplace in invalid_inplaces:
+            with self.assertRaisesRegex(
+                    ValueError,
+                    'For argument "inplace" expected type bool, received type {}.'
+                    .format(type(inplace).__name__)):
+                kdf.query('a < b', inplace=inplace)
+
+        # doesn't support for MultiIndex columns
+        columns = pd.MultiIndex.from_tuples([('A', 'Z'), ('B', 'X'), ('C', 'C')])
+        kdf.columns = columns
+        with self.assertRaisesRegex(ValueError, "Doesn't support for MultiIndex columns"):
+            kdf.query("('A', 'Z') > ('B', 'X')")
```

Comment:
Wait, is it right syntax in pandas? Seems pandas [doesn't support multi-index column yet] (https://github.com/pandas-dev/pandas/issues/13348)

Reply:
they're supporting both

```python
>>> d
  L1
   A  B
0  0  0
1  4  1
2  4  1
3  0  3

>>> d.L1.A
0    0
1    4
2    4
3    0
Name: A, dtype: int64

>>> d[('L1', 'A')]
0    0
1    4
2    4
3    0
Name: (L1, A), dtype: int64
```

Reply:
> Seems pandas doesn't support multi-index column yet

and yes, pandas doesn't support MultiIndex columns at least for `query`.